Tests/NIOTests/ChannelTests.swift

```diff
@@ -1442,4 +1442,88 @@ public class ChannelTests: XCTestCase {
         try clientChannel.closeFuture.wait()
         try serverChannel.syncCloseAcceptingAlreadyClosed()
     }
+
+    func testWeDontJamSocketsInANoIOState() throws {
+        final class ReadDelayer: ChannelDuplexHandler {
+            typealias InboundIn = Any
+            typealias InboundOut = Any
+            typealias OutboundIn = Any
+            typealias OutboundOut = Any
```

Comment:
@Lukasa not that it matters much but shouldn't this really be `NIOAny` ? Just for the sake of show "more correct" code to people that read it ?

Reply:
No. `InboundIn` and friends are what we cast `NIOAny` *to*: that is, they're the type that we expect to receive as the return value from `unwrapInboundIn`. We presumably don't expect `NIOAny` there.